### PR TITLE
Add the ability to edit users

### LINKED ISSUE #5: Enhance Users
- [x] Add `active` and `admin` fields
- [x] Add a `password_confirmation` field when registering
- [x] Add a registration page
- [x] Add a login page
- [x] Add an edit page
- [x] Force users to log in or register if they do not have an existing session
- [x] Allow a user to edit their own configuration
- [ ] Update tests

### PR DESCRIPTION
**Pull request by Serneum**
Only the last commit needs to be reviewed

Closes #5 

image

image

### CODE REVIEW

**Comment by Serneum** [comment]
I seem to have broken the ability to log out

File: lib/agoneum_web/router.ex [hunk 1]
```diff
@@ -7,47 +7,52 @@ defmodule AgoneumWeb.Router do
...
   pipeline :api do
     plug :accepts, ["json"]
+    plug Guardian.Plug.VerifyHeader, realm: "Bearer"
+    plug Guardian.Plug.LoadResource
   end
 
   pipeline :api_auth do
-    plug Guardian.Plug.VerifyHeader, realm: "Bearer"
-    plug Guardian.Plug.LoadResource
     plug Guardian.Plug.EnsureResource
```
Comment (Serneum) [inline lib/agoneum_web/router.ex]: Pretty sure this needs a `handler`

File: lib/agoneum_web/router.ex [hunk 2]
```diff
@@ -7,47 +7,52 @@ defmodule AgoneumWeb.Router do
     plug :fetch_flash
     plug :protect_from_forgery
     plug :put_secure_browser_headers
+    plug Guardian.Plug.VerifySession
+    plug Guardian.Plug.LoadResource
+    plug Agoneum.Plug.Authenticate
   end
 
   pipeline :browser_auth do
```
Comment (Serneum) [inline lib/agoneum_web/router.ex]: I'm not sure how well this and `api_auth` are named now

File: lib/agoneum_web/templates/layout/app.html.slim [hunk 3]
```diff
@@ -25,6 +25,7 @@ html lang="en"
                 span.caret
               ul.dropdown-menu
                 = if logged_in?(@conn) do
+                  = link "Edit", to: profile_path(@conn, :edit), class: "dropdown-item"
                   = link "Log Out", to: session_path(@conn, :delete), method: "delete", class: "dropdown-item"
```
Comment (Serneum) [inline lib/agoneum_web/templates/layout/app.html.slim:29]: This seems to have stopped working. It looks like the controller method never even gets called

**Comment by Serneum** [comment]
Pretty sure our logout issues are caused by updating phoenix_html to 2.10.x. I've [filed an issue](https://github.com/phoenixframework/phoenix_html/issues/167)

File: lib/agoneum_web/templates/layout/app.html.slim (hunk 3 above)
Comment (Serneum) [reply lib/agoneum_web/templates/layout/app.html.slim:29]: I had to delete `assets/node_modules` and rerun `npm install`

File: lib/agoneum/account/account.ex [hunk 4]
```diff
@@ -24,6 +24,22 @@ defmodule Agoneum.Account do
   @doc """
   Gets a single user.
 
+  Rerturns nil if the User does not exist.
```
Comment (mbramson) [inline lib/agoneum/account/account.ex]: Typo here

**Review by mbramson — APPROVED** [review]
LGTM

**Pull request merged**